### ISSUE TITLE
[JENKINS-20620] Memory Leak on Jenkins LTS 1.509.4/1.532.1

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#15128]
<p>After Jenkins upgraded to LTS 1.509.4 from LTS 1.509.3 I notice that over time (24 hours) Jenkins becomes very slowly. <br/>
It turns out that Jenkins (under Jetty service) slowly "eats" the server memory. It's takes about 24 hours to take all the memory allocated to Jenkins (42GB). See the snapshots with examples...</p>

<p>TEST #1 on LTS 1.509.4:<br/>
1. Machine with Jetty up after restart<br/>
2. Jenkins Used - After one Hour: 22GB<br/>
3. Jenkins Used - After 12 Hours: 27GB<br/>
4. Jenkins Used - After 20 Hours: 35GB -> Memory leaks between 10:00-10:20 as you can see after GC it's still think Java in-use and fail to cleanup the all memory as it should be.<br/>
5. Jenkins Used - After 23 Hours: 39GB -> Very slow response and Heap is almost 100%</p>

<p>TEST #2 on LTS 1.509.4:<br/>
I tried to do manual GC, Doesn't help!<br/>
(see attach file: "Monitor_Memory_Over_Time_Manual_GC")</p>

<p>TEST #3 on LTS 1.509.3:<br/>
Unfortunately I downgrade to LTS 1.509.3 because of the memory leak, for me it's a blocker issue! </p>

<p>Please note that on version LTS 1.509.3 Jenkins works stable even on high environment without any memory leak... (See attach files: "Good_GC_A1.509.3" and "Good_GC_B1.509.3") but unfortunately there is a BIG unsolved problem/bug in this version, I can't rename jobs (Deadlock! which solved on the next version LTS 1.509.4/1.532.1 that I can't use because of the memory leak).</p>

<p>TEST #4 with LTS 1.532.1:<br/>
Same issue! Jenkins stuck with 100% memory usage after only 12 hours!</p>

<p>Thank You,<br/>
Ronen.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ronenpg">ronenpg</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620" target="_blank">Memory Leak on Jenkins LTS 1.509.4/1.532.1</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev">oleg_nenashev</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 1.509.4, 1.509.4LTS, 1.532.1, 1.532.1LTS, Jenkins, LTS
<li><b>resolution</b>: Incomplete
<li><b>resolved</b>: 2014-06-09T21:30:10+00:00
<li><b>votes</b>: 21
<li><b>watchers</b>: 15
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>After Jenkins upgraded to LTS 1.509.4 from LTS 1.509.3 I notice that over time (24 hours) Jenkins becomes very slowly. 
It turns out that Jenkins (under Jetty service) slowly "eats" the server memory. It's takes about 24 hours to take all the memory allocated to Jenkins (42GB). See the snapshots with examples...</p>

<p>TEST #1 on LTS 1.509.4:
1. Machine with Jetty up after restart
2. Jenkins Used - After one Hour: 22GB
3. Jenkins Used - After 12 Hours: 27GB
4. Jenkins Used - After 20 Hours: 35GB -&gt; Memory leaks between 10:00-10:20 as you can see after GC it's still think Java in-use and fail to cleanup the all memory as it should be.
5. Jenkins Used - After 23 Hours: 39GB -&gt; Very slow response and Heap is almost 100%</p>

<p>TEST #2 on LTS 1.509.4:
I tried to do manual GC, Doesn't help!
(see attach file: "Monitor_Memory_Over_Time_Manual_GC")</p>

<p>TEST #3 on LTS 1.509.3:
Unfortunately I downgrade to LTS 1.509.3 because of the memory leak, for me it's a blocker issue! </p>

<p>Please note that on version LTS 1.509.3 Jenkins works stable even on high environment without any memory leak... (See attach files: "Good_GC_A1.509.3" and "Good_GC_B1.509.3") but unfortunately there is a BIG unsolved problem/bug in this version, I can't rename jobs (Deadlock! which solved on the next version LTS 1.509.4/1.532.1 that I can't use because of the memory leak).</p>

<p>TEST #4 with LTS 1.532.1:
Same issue! Jenkins stuck with 100% memory usage after only 12 hours!</p>

<p>Thank You,
Ronen.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Linux Server with 24 CPUs and 64GB RAM<br/>
Jenkins version LTS 1.509.4/1.532.1 on Jetty<br/>
Memory allocated for Jenkins/Jetty process: 42GB<br/>
Environment: Jenkins working with 600 jobs with high activities + 40 slave machines (Linux and Windows)
```
</details>
<details><summary><i>8 attachments</i></summary>

- [Good_GC_A1.509.3.JPG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24688/Good_GC_A1.509.3.JPG)
  > ![Good_GC_A1.509.3.JPG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24688/Good_GC_A1.509.3.JPG)
- [Good_GC_B1.509.3.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24766/Good_GC_B1.509.3.jpg)
  > ![Good_GC_B1.509.3.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24766/Good_GC_B1.509.3.jpg)
- [Monitor_After_12_Hours.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24682/Monitor_After_12_Hours.jpg)
  > ![Monitor_After_12_Hours.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24682/Monitor_After_12_Hours.jpg)
- [Monitor_After_1_Hour.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24681/Monitor_After_1_Hour.jpg)
  > ![Monitor_After_1_Hour.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24681/Monitor_After_1_Hour.jpg)
- [Monitor_After_20_Hours.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24683/Monitor_After_20_Hours.jpg)
  > ![Monitor_After_20_Hours.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24683/Monitor_After_20_Hours.jpg)
- [Monitor_After_23_Hours.JPG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24684/Monitor_After_23_Hours.JPG)
  > ![Monitor_After_23_Hours.JPG](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24684/Monitor_After_23_Hours.JPG)
- [Monitor_Memory_Over_Time_Manual_GC.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24685/Monitor_Memory_Over_Time_Manual_GC.jpg)
  > ![Monitor_Memory_Over_Time_Manual_GC.jpg](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24685/Monitor_Memory_Over_Time_Manual_GC.jpg)
- [gc.log](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/24686/gc.log)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>elyf:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=190858&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-190858">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This same problem happens with 1.540 and 1.541 on a red hat Linux platform </p>
</pre>
</details>

</ul>
</details>

<p>This same problem happens with 1.540 and 1.541 on a red hat Linux platform </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=190900&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-190900">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The same problem happens with 1.532.1 LTS !</p>
</pre>
</details>

</ul>
</details>

<p>The same problem happens with 1.532.1 LTS !</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tomerpg.png" title="tomerpg's avatar" /> tomerpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=190946&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-190946">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The same problem with memory leak on our system with 1.509.4 LTS, 
After 8 hours the Jenkins become very slowly and then stuck, I must do restart to Jenkins every day!</p>
</pre>
</details>

</ul>
</details>

<p>The same problem with memory leak on our system with 1.509.4 LTS, <br/>
After 8 hours the Jenkins become very slowly and then stuck, I must do restart to Jenkins every day!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>vjuranek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191144&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191144">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi,
any change to see some heap dump (preferably some small one), or, as a first hint (as I won't be probably able to parse 40GB heap dump on my machine anyway), see output of</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-none">jmap -histo:live $JENKINS_PID |head -n 30
</pre>
</div></div>
<p>?
Thanks</p>
</pre>
</details>

</ul>
</details>

<p>Hi,<br/>
any change to see some heap dump (preferably some small one), or, as a first hint (as I won't be probably able to parse 40GB heap dump on my machine anyway), see output of</p>

<pre>
jmap -histo:live $JENKINS_PID |head -n 30
</pre>
<p>?<br/>
Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191451&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191451">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry "vjuranek" already returns to the old and stable LTS version: 1.509.3 - The version LTS 1.509.3 come with a lots known bugs but with free of memory leaks. </p>

<p>Any Jenkins system with more than 50 jobs can see very fast the memory leaks on versions LTS 1.509.4 or 1.532.1, I don't understand why Jenkins owners ignore this very blocked issue... maybe Jenkins support team need a Load Software to find their own bugs and issues.</p>
</pre>
</details>

</ul>
</details>

<p>Sorry "vjuranek" already returns to the old and stable LTS version: 1.509.3 - The version LTS 1.509.3 come with a lots known bugs but with free of memory leaks. </p>

<p>Any Jenkins system with more than 50 jobs can see very fast the memory leaks on versions LTS 1.509.4 or 1.532.1, I don't understand why Jenkins owners ignore this very blocked issue... maybe Jenkins support team need a Load Software to find their own bugs and issues.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>vjuranek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191453&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191453">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It seems that JoJ <span class="error">&#91;1&#93;</span> runs without any problems, so this memory leak maybe appears only on some specific configurations (or somehow fixed in 1.539?). Did you only upgrade Jenkins core or did you also upgrade any plugins (so we can exclude memory leak in the plugins)? By system with 50 jobs you mean 50 jobs present or 50 jobs run in parallel most of the time?
Thanks</p>

<p>Btw: there's no Jenkins support team, it's community project based on volunteers, if you want a support team, you should check Cloudbees or some other provider which offers support for Jenkins. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p><span class="error">&#91;1&#93;</span> <a href="https://ci.jenkins-ci.org/" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins-ci.org/</a></p>
</pre>
</details>

</ul>
</details>

<p>It seems that JoJ <span class="error">&#91;1&#93;</span> runs without any problems, so this memory leak maybe appears only on some specific configurations (or somehow fixed in 1.539?). Did you only upgrade Jenkins core or did you also upgrade any plugins (so we can exclude memory leak in the plugins)? By system with 50 jobs you mean 50 jobs present or 50 jobs run in parallel most of the time?<br/>
Thanks</p>

<p>Btw: there's no Jenkins support team, it's community project based on volunteers, if you want a support team, you should check Cloudbees or some other provider which offers support for Jenkins. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p><span class="error">&#91;1&#93;</span> <a href="https://ci.jenkins-ci.org/" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins-ci.org/</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191455&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191455">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thank You for your help <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>First with all the respect to JoJ they doesn't have the system load that we are facing here and has you can see it's snapshots version. Anyway, I didn't try the release version 1.539... BTW: In order to solve memory leaks you must use load software such as Load Runner.</p>

<p>I tried the new LTS versions after update plugins and without update plugins, still the same issue same bug and as I wrote this bug doesn't happens on previous version at all.</p>

<p>Of course it community, open source and all of that... but don't forget there is also Enterprise edition that including "Jenkins support team" and I just wonder why they not solving this super critical issue but I guess this all another story...</p>
</pre>
</details>

</ul>
</details>

<p>Thank You for your help <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>First with all the respect to JoJ they doesn't have the system load that we are facing here and has you can see it's snapshots version. Anyway, I didn't try the release version 1.539... BTW: In order to solve memory leaks you must use load software such as Load Runner.</p>

<p>I tried the new LTS versions after update plugins and without update plugins, still the same issue same bug and as I wrote this bug doesn't happens on previous version at all.</p>

<p>Of course it community, open source and all of that... but don't forget there is also Enterprise edition that including "Jenkins support team" and I just wonder why they not solving this super critical issue but I guess this all another story...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191924&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191924">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Update: I checked the release version: 1.539 that on JoJ, this version is memory leaks free but this version suffers from other bugs such as problem when using Build multi-configuration project (Matrix job) + Builds of a concurrently executable job.</p>
</pre>
</details>

</ul>
</details>

<p>Update: I checked the release version: 1.539 that on JoJ, this version is memory leaks free but this version suffers from other bugs such as problem when using Build multi-configuration project (Matrix job) + Builds of a concurrently executable job.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191968&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191968">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Update: I checked also the 1.54X versions (1.541, 1.542, 1.543, 1.544) and the memory leak issue is also on all release 1.54X versions!</p>
</pre>
</details>

</ul>
</details>

<p>Update: I checked also the 1.54X versions (1.541, 1.542, 1.543, 1.544) and the memory leak issue is also on all release 1.54X versions!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191970&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191970">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Could you collect a heap dump / any other memory statistics of your system?
We use 1.509.4 with a quite similar configuration, but we have not experienced such memory leaks (the uptime is close to 3 months). There are many other issues, but seems that the memory is OK...</p>
</pre>
</details>

</ul>
</details>

<p>Could you collect a heap dump / any other memory statistics of your system?<br/>
We use 1.509.4 with a quite similar configuration, but we have not experienced such memory leaks (the uptime is close to 3 months). There are many other issues, but seems that the memory is OK...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191976&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191976">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks Oleg.</p>

<p>I can't dump it because it's to big (40GB) and for some reason the "jmap" unable to connect to the Jenkins process. Anyway I downgraded the version to 1.537...</p>

<p>Oleg, If you have a system with 600 jobs with high activities + 40 slave machines online and you don't have this issue it's very strange because I test these versions on several different machines and each time the problem reproduced (after 12-24 hours).</p>
</pre>
</details>

</ul>
</details>

<p>Thanks Oleg.</p>

<p>I can't dump it because it's to big (40GB) and for some reason the "jmap" unable to connect to the Jenkins process. Anyway I downgraded the version to 1.537...</p>

<p>Oleg, If you have a system with 600 jobs with high activities + 40 slave machines online and you don't have this issue it's very strange because I test these versions on several different machines and each time the problem reproduced (after 12-24 hours).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191977&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191977">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Probably, we have different job contents/plugins (e.g. we don't use Maven plugin; xUnit components are not popular due to integration with external systems).
Could you provide a list of your plugins? Since we know about stable/unstable versions, we can try to find an interoperability issue if it exists. </p>


</pre>
</details>

</ul>
</details>

<p>Probably, we have different job contents/plugins (e.g. we don't use Maven plugin; xUnit components are not popular due to integration with external systems).<br/>
Could you provide a list of your plugins? Since we know about stable/unstable versions, we can try to find an interoperability issue if it exists. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=191979&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-191979">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Oleg, I sent you email with our Jenkins Plugins list. Thank You <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Oleg, I sent you email with our Jenkins Plugins list. Thank You <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nickolayr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=193221&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-193221">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>BTW, Ronen, do you use System Groovy script buildsteps in your jobs?</p>
</pre>
</details>

</ul>
</details>

<p>BTW, Ronen, do you use System Groovy script buildsteps in your jobs?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=193402&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-193402">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Nickolay, Yes we have a Groovy scripts on our jobs.</p>
</pre>
</details>

</ul>
</details>

<p>Hi Nickolay, Yes we have a Groovy scripts on our jobs.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=194696&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-194696">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Update: 
The solution was to delete some slave machines from the Jenkins nodes. 
It turns out that Jenkins can't handle more than 100 slave machines. 
Currently we have (after cleanup) 70 slave machines and no memory leak!</p>

<p>BTW: The memory leak issue occurs only on Jenkins Master running on Linux O/S with more than 100 slave machines, Actually on Windows O/S this issue doesn't exist!</p>
</pre>
</details>

</ul>
</details>

<p>Update: <br/>
The solution was to delete some slave machines from the Jenkins nodes. <br/>
It turns out that Jenkins can't handle more than 100 slave machines. <br/>
Currently we have (after cleanup) 70 slave machines and no memory leak!</p>

<p>BTW: The memory leak issue occurs only on Jenkins Master running on Linux O/S with more than 100 slave machines, Actually on Windows O/S this issue doesn't exist!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=200034&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200034">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've tested Jenkins 1.509.4(patched by remoting-2.36)/RHEL6.4 with about 150 slaves.
There's no memory leak after 1 week. The test installation just builds several Jenkins plugins, hence there's no extremal load</p>

<p>Probably, the error could be in the communication layer. I'll try the remoting version from 1.532.1 with a bigger workload</p>
</pre>
</details>

</ul>
</details>

<p>I've tested Jenkins 1.509.4(patched by remoting-2.36)/RHEL6.4 with about 150 slaves.<br/>
There's no memory leak after 1 week. The test installation just builds several Jenkins plugins, hence there's no extremal load</p>

<p>Probably, the error could be in the communication layer. I'll try the remoting version from 1.532.1 with a bigger workload</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=203279&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203279">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We need more information to be able to solve problems like this. Please see <a href="https://wiki.jenkins-ci.org/display/JENKINS/I%27m+getting+OutOfMemoryError" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/I%27m+getting+OutOfMemoryError</a> for how to get the details we need to be able to work on problems like this.</p>

<p>I'm not doubting that you are seeing the problem, and for that I am sorry. Please get us the details we need so that we can fix the problem.</p>

<p>If you cannot post a heap dump, please get at least the histogram summary.</p>
</pre>
</details>

</ul>
</details>

<p>We need more information to be able to solve problems like this. Please see <a href="https://wiki.jenkins-ci.org/display/JENKINS/I%27m+getting+OutOfMemoryError" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/I%27m+getting+OutOfMemoryError</a> for how to get the details we need to be able to work on problems like this.</p>

<p>I'm not doubting that you are seeing the problem, and for that I am sorry. Please get us the details we need so that we can fix the problem.</p>

<p>If you cannot post a heap dump, please get at least the histogram summary.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ronenpg.png" title="ronenpg's avatar" /> ronenpg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-20620?focusedId=223531&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-223531">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@Oleg Nenashev, Did you try it with 1200 active jobs? anyway this is what solved my problem. I guess you have issue with 100+ slave machines connected to Jenkins with high load Jenkins.</p>

<p>@Kohsuke Kawaguchi, because I have 64GB RAM, I can't do it, I can't save 64GB RAM on my HDD and anyway my problem is already solved so it's save to close it.</p>
</pre>
</details>

</ul>
</details>

<p>@​Oleg Nenashev, Did you try it with 1200 active jobs? anyway this is what solved my problem. I guess you have issue with 100+ slave machines connected to Jenkins with high load Jenkins.</p>

<p>@​Kohsuke Kawaguchi, because I have 64GB RAM, I can't do it, I can't save 64GB RAM on my HDD and anyway my problem is already solved so it's save to close it.</p>
